### PR TITLE
Message Fixes

### PR DESCRIPTION
**Pull request by Jedzkie**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
In official, skill/item messages when used in a restricted map will show cyan color. In our current system, it uses white color.

[//]: # (Describe at length, the changes that this pull request makes.)

**Affected Branches:** 
master

[//]: # (Master? Slave?)

**Issues addressed:**

[//]: # (Issue Tracker Number if any.)

### Known Issues and TODO List

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2006)

File: npc/mapflag/noteleport.txt [hunk 1]
```diff
@@ -154,6 +154,7 @@ thor_camp	mapflag	noteleport
 moc_para01	mapflag	noteleport
 
 // Job Quests ====================
+pay_arche	mapflag	noteleport
```
Comment (4144) [inline npc/mapflag/noteleport.txt]: this change look like unrelated to messages fixes

File: src/map/clif.h [hunk 2]
```diff
@@ -369,8 +369,6 @@ enum clif_messages {
 	MSG_MERCENARY_ESCAPED          = 0x4f5, ///< The mercenary has run away.
 	MSG_PARTY_MEMBER_NOT_SUMMONED  = 0x4c5, ///< The party member was not summoned because you are not the party leader.
 	MSG_PARTY_NO_MEMBER_IN_MAP     = 0x4c6, ///< There is no party member to summon in the current map.
-	MSG_SKILL_CANT_USE_AREA        = 0x536, ///< This skill cannot be used within this area
```
Comment (4144) [inline src/map/clif.h]: remove known messages from msgstringtable probably not good idea.

File: src/map/clif.c [hunk 3]
```diff
@@ -5466,6 +5466,7 @@ void clif_skill_warppoint(struct map_session_data* sd, uint16 skill_id, uint16 s
 ///     0 = "Saved location as a Memo Point for Warp skill." in color 0xFFFF00 (cyan)
 ///     1 = "Skill Level is not high enough." in color 0x0000FF (red)
 ///     2 = "You haven't learned Warp." in color 0x0000FF (red)
+///     3 = "Cannot save location as a Memo Point at current location." in color 0x0000FF (red)
```
Comment (4144) [inline src/map/clif.c]: is old clients support this flag too?

File: src/map/clif.c [hunk 4]
```diff
@@ -5488,6 +5489,7 @@ void clif_skill_memomessage(struct map_session_data* sd, int type)
 ///     0 = "Unable to Teleport in this area" in color 0xFFFF00 (cyan)
 ///     1 = "Saved point cannot be memorized." in color 0x0000FF (red)
 ///     2 = "This skill cannot be used within this area." in color 0xFFFF00 (cyan)
+///     3 = "This item cannot be used within this area." in color 0xFFFF00 (cyan)
```
Comment (4144) [inline src/map/clif.c]: is old clients support this flag too?

File: npc/mapflag/noteleport.txt (hunk 1 above)
Comment (4144) [reply npc/mapflag/noteleport.txt]: if you want add map flags, do this in separate commit

File: src/map/clif.h (hunk 2 above)
Comment (Jedzkie) [reply src/map/clif.h]: This message is useless now, since I already know the way how to show this properly.

File: src/map/clif.c (hunk 4 above)
Comment (Jedzkie) [reply src/map/clif.c]: Fixed by adding the client provided by you, also I added a message in message.conf

File: npc/mapflag/noteleport.txt (hunk 1 above)
Comment (Jedzkie) [reply npc/mapflag/noteleport.txt]: Fixed.

File: src/map/clif.h (hunk 2 above)
Comment (4144) [reply src/map/clif.h]: better left it for future usage.
more constants added here is better. if even unused now.
for old clients you can show this message from msgstringtable

File: src/map/pc.c [hunk 5]
```diff
@@ -4943,12 +4949,19 @@ int pc_isUseitem(struct map_session_data *sd,int n)
 		}
 		FALLTHROUGH
 		case ITEMID_WING_OF_FLY:
+		case ITEMID_N_FLY_WING:
+		case ITEMID_C_WING_OF_FLY:
 			if (map->list[sd->bl.m].flag.noteleport || map_flag_gvg2(sd->bl.m)) {
-				clif->skill_mapinfomessage(sd, 0);
+#if PACKETVER >= 20080311
+				clif->skill_mapinfomessage(sd, 3);
```
[the remark this answers is not in the document]
Comment (Jedzkie) [reply src/map/pc.c]: How about other items? Like Neuralizer and Butterfly Wings?

[the remark this answers is not in the document]
Comment (Jedzkie) [reply src/map/pc.c]: in iRO, the message is shown cyan color.
